producer-c/producer-cloudwatch-integ/jobs/runner.groovy

```diff
@@ -0,0 +1,251 @@
+import jenkins.model.*
+import org.jenkinsci.plugins.workflow.steps.FlowInterruptedException
+
+START_TIMESTAMP = new Date().getTime()
+HAS_ERROR = false
+
+RUNNING_NODES=0
+
+def buildProducer() {
+  sh  """
+    cd $WORKSPACE/producer-c/producer-cloudwatch-integ && 
+    if [ ! -d "build" ] 
+    then
+        mkdir build
+    fi
+    cd build && 
+    sudo cmake .. && 
+    sudo make 
+  """
+}
+
+def buildConsumer() {
+  sh '''
+    JAVA_HOME='/opt/jdk-13.0.1'
+    PATH="$JAVA_HOME/bin:$PATH"
+    export PATH
+    M2_HOME='/opt/apache-maven-3.6.3'
+    PATH="$M2_HOME/bin:$PATH"
+    export PATH
+    cd $WORKSPACE/consumer-java/aws-kinesis-video-producer-sdk-canary-consumer
+    mvn package
+    touch tmp_jar
+    mvn -Dmdep.outputFile=tmp_jar dependency:build-classpath
+  '''
+}
+
+def runProducer(envs) {
+    def credentials = [
+        [
+            $class: 'AmazonWebServicesCredentialsBinding', 
+            accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+            credentialsId: 'SDK_CANARY_CREDS',
+            secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'
+        ]
+    ]
+    withEnv(envs) {
+        withCredentials(credentials) {
+            try {
+                sh """
+                    cd $WORKSPACE/producer-c/producer-cloudwatch-integ/build && 
+                    ./kvsProducerSampleCloudwatch
+                """
+            } catch (FlowInterruptedException err) {
+                echo 'Aborted producer due to cancellation'
+                throw err
+            } catch (err) {
+                HAS_ERROR = true
+                // Ignore errors so that we can auto recover by retrying
+                unstable err.toString()
+            }
+        }
+    }
+}
+
+def runConsumer(envs) {
+    def credentials = [
+        [
+            $class: 'AmazonWebServicesCredentialsBinding', 
+            accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+            credentialsId: 'SDK_CANARY_CREDS',
+            secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'
+        ]
+    ]
+    
+    withEnv(envs) {
+        withCredentials(credentials) {
+             try {
+                sh '''
+                    echo "Building comsumer"
+                    JAVA_HOME='/opt/jdk-13.0.1'
```

Comment:
I think it's cleaner to use groovy to populate the env variables. For `PATH`, I would populate another env variable and append it to `PATH`, e.g. `export PATH=$CANARY_PATH:$PATH`.

Also, you don't need to separate `PATH` from `export PATH`, you can do this in one line, `export PATH=...`